### PR TITLE
[Bulky Goods] Add page for viewing existing booking

### PR DESCRIPTION
**Pull request by davea**
For https://github.com/mysociety/societyworks/issues/3373

(will sort out photo storage in separate PR)

[skip changelog]

### CODE REVIEW

**Review by dracos — CHANGES_REQUESTED** [review]
Looks good (as you say in comment, can probably combine bits with the payment back PR for tidiness). However, is there a reason the view booking page can't be instead of the /report/ID page - do we need two 'views' on the same 'report'? This would also then answer my next problem, which is that the confirmation/reminder emails (and other places?) with links to cancel, I think they should probably go to the relevant view booking page, which I assume would be easier if it was to `/report/ID` as we don't have the postcode to construct the relevant page for this PR. If there's a reason we can't do that (let me know), I think we'll have to store the `property_id` with the report (I think this is straightforward, setting `property_id` as extra data on the category).

**Comment by davea** [comment]
🤦🏻‍♂️I don't know why I didn't think of that. Great idea, will update accordingly.

File: perllib/FixMyStreet/Cobrand/Peterborough.pm [hunk 1]
```diff
@@ -1130,6 +1144,25 @@ sub waste_munge_bulky_data {
     # XXX what about photos?
 }
 
+sub waste_reconstruct_bulky_data {
```
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/Peterborough.pm:1147]: This should hopefully disappear with a rebase now.

File: perllib/FixMyStreet/Cobrand/Peterborough.pm [hunk 2]
```diff
@@ -514,9 +514,23 @@ sub has_pending_bulky_collection {
             state =>
                 { '=', [ FixMyStreet::DB::Result::Problem->open_states ] },
         },
-    )->all;
+    );
+}
+
+sub bulky_can_view_collection {
```
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/Peterborough.pm:520]: Do we need this function at all now, or is the Report.pm code doing it for us already, it already has all the 'private = reporter or staff only' checks in.

File: perllib/FixMyStreet/App/Controller/Waste.pm [hunk 3]
```diff
@@ -998,6 +997,11 @@ sub bulky : Chained('bulky_setup') : Args(0) {
     $c->stash->{first_page} = 'intro';
     $c->stash->{form_class} = 'FixMyStreet::App::Form::Waste::Bulky';
 
+    if ( $c->stash->{property}{pending_bulky_collection} ) {
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Waste.pm:1000]: Guess this can go back where it was too, though guess it doesn't matter.

**Review by dracos — CHANGES_REQUESTED** [review]
Looks good, though I think the user checking can be dropped now? Perhaps with a comment around bulky_view explaining it's being called as a report page so someone doesn't add it in future by mistake.

File: perllib/FixMyStreet/Cobrand/Peterborough.pm (hunk 2 above)
Comment (davea) [reply perllib/FixMyStreet/Cobrand/Peterborough.pm:520]: It's still being called from the `bin_days_bulky.html` template so still needed. Don't think it needs to be called from within `bulky_view` though, you're right.

**Review by dracos — APPROVED** [review]
Woo, nearly there :) (overall, not this PR ;) )

**Pull request merged**